internal/services/appconfiguration/app_configuration.go

```diff
@@ -81,3 +83,61 @@ func appConfigurationGetKeyRefreshFunc(ctx context.Context, client *appconfigura
 		return res, "Exists", nil
 	}
 }
+
+func appConfigurationGetKeyRefreshFuncForUpdate(ctx context.Context, client *appconfiguration.BaseClient, key, label string, model appconfiguration.KeyValue) pluginsdk.StateRefreshFunc {
```

Comment:
This is added to ensure the App Config Key/Feature property is fully synced after an update.